torch/nn/utils/stateless.py

```diff
@@ -105,6 +140,7 @@ def functional_call(
     parameters_and_buffers: Dict[str, Tensor],
     args: Union[Any, Tuple],
     kwargs: Dict[str, Any] = None,
+    tie_weights: bool = False,
```

Comment:
nit: "weights" can refer to only parameters, "tie_weights" effectively ties both parameters and buffers. Is that a problem? If we think it's a problem, we can rename it tie_weights_and_buffers. I kind of like the shorter name (and we can document that it applies to both parameters and buffers), but open to suggestions.

Docstring for nn.utils.stateless.functional_call needs to be updated with new flag and details on what it does (unless the plan was to just update torch.func.functional_call)

Reply:
I'm somewhat attached to just weights since the [original paper](https://paperswithcode.com/method/weight-tying) calls it weight tying (granted there they are only tying the weight/parameters). However I hear you that it's ambiguous so the docstring comment explicitly calls out that it ties both parameters and buffers

Reply:
SGTM